server/embed/config.go

```diff
@@ -156,6 +156,46 @@ func init() {
 	defaultHostname, defaultHostStatus = netutil.GetDefaultHost()
 }
 
+type Duration struct {
+	time.Duration
+}
+
+func FromTimeDuration(duration time.Duration) Duration {
+	return Duration{Duration: duration}
+}
+
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var v any
+	var err error
+	if err = yaml.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		d.Duration = time.Duration(value)
+		return nil
+	case string:
+		d.Duration, err = time.ParseDuration(value)
+
+		return err
+	}
+	return err
+}
+
+func (d Duration) String() string {
+	return d.Duration.String()
+}
```

Comment:
And this is not required, either. As it would call `String()` from `d.Duration`.